File: atom-2024-front/src/app/models/UserDto.ts
import {RoleDto} from './RoleDto';
import {DateFormatter} from '../components/common/DateFormatter';

export class UserDto {
  id: UserId;
  firstname: string;
  lastname: string;
  email: string;
  password: string;
  registrationDate: Date = new Date();
  roles: RoleDto[];
  surname?: string;
  phoneNumber?: string;
  archived: boolean;
  fullName: string;
  shortName: string;
  rolesAsString: string;
  tutorId?: number;
  studentId?: number;

  get formattedRegistrationDate() {
    return DateFormatter.format(this.registrationDate);
  }

  constructor(source?: any) {
    if (source) {
      Object.assign(this, source);
    }
  }
}

export type UserId = number;

File: atom-2024-backend/src/main/java/com/prolegacy/atom2024backend/common/auth/dto/UserDto.java
package com.prolegacy.atom2024backend.common.auth.dto;

import com.prolegacy.atom2024backend.common.auth.entities.id.UserId;
import com.prolegacy.atom2024backend.entities.ids.StudentId;
import com.prolegacy.atom2024backend.entities.ids.TutorId;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.Set;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder(toBuilder = true)
public class UserDto {
    private UserId id;
    private String email;
    private String password;
    private String firstname;
    private String lastname;
    private String surname;
    private String phoneNumber;
    private Instant registrationDate;
    private Boolean archived;

    private TutorId tutorId;
    private StudentId studentId;

    private String fullName;
    private String shortName;
    private String rolesAsString;

    private Set<RoleDto> roles;
}
